plugins/defaultplugins/plugin_api_vpp.go

```diff
@@ -80,4 +81,16 @@ type API interface {
 
 	// DumpACL returns a list of all configured ACLs.
 	DumpACL() (acls []*acl.AccessLists_Acl, err error)
+
+	// DumpNat44AddressPool returns a list of all configured NAT address pools
+	DumpNat44AddressPool() ([]*vppdump.Nat44AddressPool, error)
+
+	// DumpNat44Interfaces returns a list of all enabled interfaces
+	DumpNat44Interfaces() ([]*vppdump.Nat44Interface, error)
+
+	// DumpNat44StaticMappings returns a list of static mappings
+	DumpNat44StaticMappings() ([]*vppdump.Nat44StaticMappingEntry, error)
+
+	// DumpNat44LbStaticMappings returns a list of static mappings with load balancer
+	DumpNat44LbStaticMappings() ([]*vppdump.Nat44StaticMappingEntry, error)
```

Comment:
It would be ideal if dump method has returned data in the same format as the northbound API has, i.e. instance of Nat44Global and a list of DNatConfig instances, something like:

DumpNat44GlobalConfig() (*Nat44Global, error)
DumpNat44DNATConfig() ([]*DNatConfig, error)

Otherwise we will not be able to use the plugin to the full extent in contiv/vpp and much of the code that calculates diff between current and new configuration would be replicated.

In the latest VPP there is now a VPP-opaque 64-character length tag added to each mapping. It can be therefore used to store DNatConfig label which will then allow to reconstruct DNatConfig-s from the dump of NAT mapping.